python/vegafusion/vegafusion/dataset/snowpark.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import pyarrow as pa
+from .sql import SqlDataset
+from snowflake.snowpark import Table as SnowparkTable
+from snowflake.snowpark.types import DataType as SnowparkDataType
+from typing import Dict
+
+from ..transformer import to_arrow_table
+
+SNOWPARK_TO_PYARROW_TYPES: Dict[SnowparkDataType, pa.DataType] = {}
+
+
+def get_snowpark_to_pyarrow_types():
+    if not SNOWPARK_TO_PYARROW_TYPES:
+        import snowflake.snowpark.types as sp_types
+
+        SNOWPARK_TO_PYARROW_TYPES.update(
+            {
+                sp_types.LongType: pa.int64(),
+                sp_types.BinaryType: pa.binary(),
+                sp_types.BooleanType: pa.bool_(),
+                sp_types.ByteType: pa.int8(),
+                sp_types.StringType: pa.string(),
+                sp_types.DateType: pa.date32(),
+                sp_types.DoubleType: pa.float64(),
+                sp_types.FloatType: pa.float32(),
+                sp_types.IntegerType: pa.int32(),
+                sp_types.ShortType: pa.int16(),
+                sp_types.TimestampType: pa.timestamp("ms"),
+            }
+        )
+    return SNOWPARK_TO_PYARROW_TYPES
```

Comment:
@jonmmease given we don't mind importing PyArrow or Snowflake here, this could be statically initialised rather than lazily now.